node/src/accountant/mod.rs

```diff
@@ -3291,91 +3149,237 @@ mod tests {
+        message: ReportServicesConsumedMessage,
+    ) -> Arc<Mutex<Vec<(SystemTime, Wallet, u64)>>> {
         let more_money_payable_parameters_arc = Arc::new(Mutex::new(vec![]));
         let payable_dao_mock = PayableDaoMock::new()
             .non_pending_payables_result(vec![])
-            .more_money_payable_parameters(more_money_payable_parameters_arc.clone());
+            .more_money_payable_result(Ok(()))
+            .more_money_payable_params(more_money_payable_parameters_arc.clone());
         let subject = AccountantBuilder::default()
             .bootstrapper_config(config)
             .payable_dao(payable_dao_mock)
             .build();
-        let system = System::new("report_exit_service_consumed_message_is_received");
+        let system = System::new("test");
         let subject_addr: Addr<Accountant> = subject.start();
         subject_addr
             .try_send(BindMessage {
                 peer_actors: peer_actors_builder().build(),
             })
             .unwrap();
 
-        subject_addr
-            .try_send(ReportExitServiceConsumedMessage {
-                timestamp: SystemTime::now(),
+        subject_addr.try_send(message).unwrap();
+
+        System::current().stop();
+        system.run();
+        more_money_payable_parameters_arc
+    }
+
+    #[test]
+    fn report_routing_service_consumed_message_is_received_for_our_consuming_wallet() {
+        init_test_logging();
+        let consuming_wallet = make_wallet("the consuming wallet");
+        let config = bc_from_ac_plus_wallets(
+            make_populated_accountant_config_with_defaults(),
+            consuming_wallet.clone(),
+            make_wallet("the earning wallet"),
+        );
+        let foreign_wallet = make_wallet("exit wallet");
+        let timestamp = SystemTime::now();
+        let report_message = ReportServicesConsumedMessage {
+            timestamp,
+            exit: ExitServiceConsumed {
+                earning_wallet: foreign_wallet.clone(),
+                payload_size: 1234,
+                service_rate: 45,
+                byte_rate: 10,
+            },
+            routing_payload_size: 3333,
+            routing: vec![RoutingServiceConsumed {
+                earning_wallet: consuming_wallet.clone(),
+                service_rate: 42,
+                byte_rate: 6,
+            }],
+        };
+
+        let more_money_payable_params_arc =
+            do_we_ignore_own_wallets_when_recording_consumed_services(config, report_message);
+
+        let more_money_payable_params = more_money_payable_params_arc.lock().unwrap();
+        assert_eq!(
+            *more_money_payable_params,
+            //except processing the exit service there was no change in payables
+            vec![(timestamp, foreign_wallet, (45 + 10 * 1234))]
+        );
+        TestLogHandler::new().exists_log_containing(&format!(
+            "INFO: Accountant: Not recording service consumed to our wallet {}",
```

Comment:
This message is confusing. Perhaps a better one would be, "Declining to charge our wallet {} for service we provided."